src/tlog/verify.ts

```diff
@@ -85,89 +87,152 @@ export function verifyTLogIntegratedTime(bundle: Bundle): void {
   });
 }
 
-// Returns a properly formatted "VerificationPayload" for one of the
-// transaction log entires in the given bundle which can be used for SET
-// verification.
-function toVerificationPayload(
-  bundle: Bundle,
-  publicKey: string,
-  index = 0
-): VerificationPayload {
-  // Ensure bundle has tlog entries
-  const entries = bundle.verificationData?.tlogEntries;
-  if (!entries || entries.length - 1 < index) {
-    throw new Error('No tlog entries found in bundle');
+export function verifyTLogBodies(bundle: Bundle): void {
+  bundle.verificationData?.tlogEntries.forEach((entry) => {
+    verifyTLogBody(entry, bundle);
+  });
+}
+
+// Compare the given intoto tlog entry to the given bundle
+function verifyTLogBody(entry: TransparencyLogEntry, bundle: Bundle): void {
+  if (!entry.kindVersion) {
+    throw new Error('no kindVersion found in bundle');
   }
 
-  // Rekor metadata
-  const { integratedTime, logIndex, logId } = entries[index];
+  const { kind, version } = entry.kindVersion;
+  const body: EntryKind = JSON.parse(entry.canonicalizedBody.toString('utf8'));
 
-  if (!logId) {
-    throw new Error('No logId found in bundle');
+  if (kind !== body.kind || version !== body.apiVersion) {
```

Comment:
All these consistency checks also don't feel great but better than the non-standard SET hash calculation.. trust is hard! 😅